### PR TITLE
wrapper for CUDA kernel of `CausalConv1D`

### PR DESCRIPTION
**Pull request by OliviaViessmann**
pytorch wrapper that replaces causal (padding) + torch.nn.Conv1D implementations. 

The wrapper is around Ali's subQ-ops cuda implementation. 
Wrapper is in `ops` directory in source code:  `nvsubquadratic/ops/short_causalconv1d_kernel.py`

Tests are for causality (no leakage) and parity to native pytorch implementation. 
Tests are in `tests/short_causalconv1d_kernel.py` 

Benchmark is solely for speed comparison between cuda and pytorch versions for different convolutional kernel sizes and sequence lengths against wall clock time in form of a png plot. The benchmark script also prints accuracy for dtypes o bf16, fp16 and fp32 to terminal. 
Benchmark is in `benchmarks/short_causalconv1d_kernel/`

### CODE REVIEW

**Comment by dwromero** [comment]
@OliviaViessmann , could you try to implement the multipath stuff we discussed, s.t., there's no case on which we have regressions?
